pkg/server/server.go

```diff
@@ -131,12 +130,33 @@ type Server struct {
 	autoUpdateExitCode    int
 }
 
-func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID string, packageManager *gpud_manager.Manager, opts ...gpud_config.OpOption) (_ *Server, retErr error) {
-	options := &gpud_config.Op{}
-	if err := options.ApplyOpts(opts); err != nil {
-		return nil, err
-	}
+type createComponentFunc func(context.Context, *lepconfig.Config) components.Component
+
+var createComponentFuncs = map[string]createComponentFunc{
+	containerd_pod_id.Name: func(ctx context.Context, _ *lepconfig.Config) components.Component {
+		exists := containerd_pod.CheckContainerdRunning(ctx)
+		if !exists {
+			return nil
+		}
```

Comment:
don't check here, move check into New function if need, and if doesn't support it, should mock a dummy or should return a special error like `Unsupported`..

We should hide some information that components need, otherwise this information will be leaked everywhere.